stumpy/mstump.py

```diff
@@ -178,14 +215,27 @@ def subspace(
     T = core._preprocess(T)
     core.check_window_size(m, max_size=T.shape[-1])
 
+    if T_subseq_isconstant is None or callable(T_subseq_isconstant):
```

Comment:
If this region of code is identical and repeated in multiple functions then we should refactor and move it into `core.py`

```
def process_multi_isconstant(T, m, T_subseq_isconstant):
    if T_subseq_isconstant is None or callable(T_subseq_isconstant):
        T_subseq_isconstant = [T_subseq_isconstant] * T.shape[0]

    T_subseq_isconstant = np.array(
        [
            core.rolling_isconstant(T[i], m, T_subseq_isconstant[i])
            for i in range(T.shape[0])
        ]
    )

    return T_subseq_isconstant
```

Reply:
This will make it easier to maintain in the future as we'd only need to fix/change it in one place

Reply:
Also, do we need to handle `fix_isconstant_isfinite_conflicts` too?

Reply:
> Also, do we need to handle fix_isconstant_isfinite_conflicts too?

I think so. If a user provides a 2D numpy boolean array, it should be checked.  I may go back and check all modules that use `T_x_subseq_isconstant` and see if they need to handle it or not.

Reply:
@seanlaw 
I was wondering if it would be a good idea to add the function `process_isconstant` that supports BOTH 1D and 2D (multi-) time series data, and (as the name may suggest) apply `fix_isconstant_isfinite_conflicts` too.

So, I have something like this in mind:

```
def process_isconstant(T, m, T_subseq_isconstant):
    """docstring"""
    ndim = T.ndim

    T = np.atleast_2d(T)
    if T_subseq_isconstant is None or callable(T_subseq_isconstant):
        T_subseq_isconstant = [T_subseq_isconstant] * T.shape[0]

    T_subseq_isconstant = np.array(
        [
            rolling_isconstant(T[i], m, T_subseq_isconstant[i])
            for i in range(T.shape[0])
        ]
    )

    T_subseq_isconstant = fix_isconstant_isfinite_conflicts(T, m, T_subseq_isconstant)

    if ndim == 1:
        T_subseq_isconstant = T_subseq_isconstant.flatten()
    
    return T_subseq_isconstant

```

So, this function supports 1D and 2D time series `T` and the dimention of its output depends on the dimention of the input `T`. I remember you mentioned that you prefer to keep  "the computation of isconstant" and "the check for having conflict with isfinite" separated. I think we still have that separation. We just provide a wrapper around them so that we do not forget to check the isfinite whenever we use `core.rolling_isconstant`.

---

For instance, in `core.preprocess`, we can do:

```
def preprocess(...):
    """docstring"""
    T = _preprocess(T, copy)
    check_window_size(m, max_size=T.shape[-1])

    T[np.isinf(T)] = np.nan

    T_subseq_isconstant = process_isconstant(T, m, T_subseq_isconstant)
    if M_T is None or Σ_T is None:
        M_T, Σ_T = compute_mean_std(T, m)
    T[np.isnan(T)] = 0
```

What do you think?

Reply:
Yep, that makes sense. Can you please:
1. Make this change (i.e., adding a `process_isconstant` function in separate issue/PR
2. Fix all PREVIOUS `isconstant` functions and submit a separate PR (if applicable)
3. Merge the above changes into this PR
4. Update this PR to leverage this change

Reply:
Thanks for the feedback! I will create a new PR to take care of this first before resuming our work here.